---
title: "Minimum Viable Analysis"
author: "R Ju"
date: "10/22/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "/Users/rju/Coursework/Phylogenetic Biology/finalproject/")
library(tidyverse)
library(ape)
library(ggtree)
```
#Data collection

To begin, I will compile a list of known Symbiodinium clade associations for species within the genus *Porites* from [coraltraits.org](coraltraits.org).

```{r data clean}

#load in coraltraits.org data and filter for relevant species and trait
por_sym <- read.csv("data/ctdb_1.1.0_data.csv") %>%
  filter(trait_name == "Symbiodinium clade" & str_detect(specie_name, "Porites")) %>%
  select(specie_id, specie_name, location_id, location_name, trait_id, trait_name, methodology_id, methodology_name, value)

#visualize data (Symbiodinium clades by species)
ggplot(por_sym) +
  geom_bar(aes(x = value, fill = value)) + 
  facet_wrap(~specie_name) + 
  labs(x = 'Symbiodinium clade',
       y = 'Observations') +
  theme_bw() + theme(panel.border = element_rect(color="black", fill=NA, size=0.75), panel.grid.major = element_blank(),
                     panel.grid.minor = element_blank(), axis.line = element_blank()) 

#visualize proportion data
por_sym %>%
  count(specie_name, value) %>%
  group_by(specie_name) %>%
  mutate(prop = prop.table(n)) %>%
  ggplot() +
  geom_col(aes(x = value, y = prop, fill = value)) + 
  facet_wrap(~specie_name) + 
  labs(x = 'Symbiodinium clade',
       y = 'Proportion of observations') +
  theme_bw() + theme(panel.border = element_rect(color="black", fill=NA, size=0.75), panel.grid.major = element_blank(),
                     panel.grid.minor = element_blank(), axis.line = element_blank()) 
```


###Build phylogenetic tree for genus Porites

For this analysis, I will build a tree using just the 18S gene and the same outgroup species (Siderastrea siderea) as the Kitano et al. 20 paper.

Clade name: Porites (taxid: 46719)

Outgroup: Siderastrea siderea (taxid: 130672)

####Using the CO1 gene

Bait sequence:
    >HM754452.1 Palythoa cf. heliodiscus SMH-2011 isolate 305.11.2 cytochrome oxidase subunit I (CO1) gene, partial cds; mitochondrial
    CAGGTAAACTTCAGGGTGACCAAAAAATCAAAAGAGATGTTGAAACAGAATCGGGTCCCCCCCTCCGGCG
    GGGTCAAAGAAAGTAGTGTTAAAATTCCTATCTGTCAACAGCATGGTTATAGCCCCGGCCAATACGGGCA
    AAGACAGTAACAATAAGAAGGCGGTGATTAAGATAGACCACACAAAGAGTGGCAATCTATCCATTGTCAT
    ACCCGGGGCCCTCATATTAAATATAGTGGTGATAAAATTCATAGCCCCTAAGATGGAAGAGACCCCGGCC
    AAGTGGAGGCTAAATATAACCATGTCCACGGCCCCCCCCGAATGAGCTTGAATGGCTGAAAGAGGCGGAT
    ACACCGTCCAACCTGTTCCTGCCCCCTGTTCCACAAAGGCGGAACCCAACAATAGAATTAGGGCGGGGGG
    CAAGAGCCAGAAACTAATGTTATTTAGTCGCGGGAAGGCCATATCTGGTGCACCGATATATAGCGGCACC
    AACCAATTCCCAAACCCCCCTATCATCACCGGCATCACCAGGAAAAAGATCATAACAAAAGCATGCGCAG
    TCACGATTACATTATAAAGATGGTCGTCCCCCAACATAGTTCCGGGAGCAGAAAGTTCTAACCTTATCAA
    CATACTGAAAGCTGTTCCTATCATACCGGCCCCAACACCAAATATTAGATATAATGTGCCAATATCTTTA
    TGATTTGTTGACCGTCGTTT

```{r}
por_phy <- read.tree("cluster/por.co1.alignment.fasta.treefile") 
por_phy$tip.label <- por_phy$tip.label %>%
  str_sub(end = sapply(str_locate_all(por_phy$tip.label,"_"), "[[", 2)-1) %>%
  str_replace("_"," ")
por_phy$tip.label[2] <- "Porites harrisoni"
por_phy$tip.label[62] <- "Porites fontanesii"

ggtree(por_phy) +
  geom_tiplab(fontface = "italic", size = 1) 

ggtree(por_phy, branch.length = "none") +
  geom_tiplab(fontface = "italic", size = 1) + 
  xlim(0,50)


plot(por_phy)
nodelabels(por_phy$node.label)

```



